lightningd/plugin.c

```diff
@@ -260,6 +278,90 @@ static struct io_plan *plugin_conn_init(struct io_conn *conn,
 	}
 }
 
+/* Callback called when parsing options. It just stores the value in
+ * the plugin_opt */
+static char *plugin_opt_set(const char *arg, struct plugin_opt *popt)
+{
+	popt->value = tal_strdup(popt, arg);
+	return NULL;
+}
+
+/* Add a single plugin option to the plugin as well as registering it with the
+ * command line options. */
+static bool plugin_opt_add(struct plugin *plugin, const char *buffer,
+			   const jsmntok_t *opt)
+{
+	const jsmntok_t *nametok, *typetok, *defaulttok, *desctok;
+	struct plugin_opt *popt;
+	nametok = json_get_member(buffer, opt, "name");
+	typetok = json_get_member(buffer, opt, "type");
+	desctok = json_get_member(buffer, opt, "description");
+	defaulttok = json_get_member(buffer, opt, "default");
+
+	if (!typetok || !nametok || !desctok) {
+		plugin_kill(plugin,
+			    "An option is missing either \"name\" or \"type\"");
+		return false;
+	}
+
+	/* FIXME(cdecker) Support numeric and boolean options as well */
+	if (!json_tok_streq(buffer, typetok, "string")) {
+		plugin_kill(plugin,
+			    "Only \"string\" options currenctly supported");
```

Comment:
typo: currently...

I wonder if we really want to separate by type?  After all, each option really is a 'string' in practice...

Reply:
Too much cryptocurrenctly in my brain :-)